### DISCUSSION TITLE
MIP4: BIP9 Upgrade Activation Time

### DISCUSSION THREAD

**Discussion by SeqSEE** [TheLindaProjectInc/MIPs#5] [MIPs]
BIP9 is a Bitcoin standard and takes ~2 weeks to activate. On Metrix the blocks required to activate a BIP9 proposal is ~2 days. The difference in blocktime between Bitcoin and Metrix was not considered when this was implemented, so it should be corrected. This proposal will change the activation time to be 1 week.

This will ensure that future BIP9 activations happen in the expected amount of time which is more reasonable for holders and miners.

[bip-0009](https://github.com/bitcoin/bips/blob/master/bip-0009.mediawiki)

[Bitcoin Core](https://github.com/bitcoin/bitcoin/blob/25202cace9140870c75cb3a811e10045df88c226/src/kernel/chainparams.cpp#L93-L94)

I've created a PR for the draft version of this MIP.

https://github.com/TheLindaProjectInc/MIPs/pull/4

**Comment by judenewcomb** [comment]
Thank you!

**Comment by squidicuzz** [comment]
I attempted to do some test runs of some code to carry out MIP4.  It seems we may not be able to use the BIP9 system to upgrade itself, as it forms an endless loop and gets complicated fast.
I am working on another possible method to complete this, possibly without using the BIP9 system to accomplish this.

**Reply by squidicuzz** [reply]
I've added a `startHeight` variable to the BIP9 system.  This will allow us to start signaling for sequential upgrades at determined block heights.

Additionally, I have tested MIP4 on `regtest` and all is well.

**Comment by SeqSEE** [comment]
MIP 4 has been activated by the network.
